libs/platform/src/lib/components/menu/menu-component/menu-item.component.html

```diff
@@ -0,0 +1,91 @@
+<ng-container
+    *ngTemplateOutlet="
+        selectable ? selectableItemTemplate : itemTemplate;
+        context: {
+            label: label,
+            i: index,
+            selected: selected,
+            item: item,
+            group: group,
+            separated: separated,
+            secondaryIcon: secondaryIcon,
+            disabled: disabled,
+            tooltipLabel: tooltipLabel,
+            itemWidth: itemWidth,
+            childItems: childItems
+        }
+    "
+>
+</ng-container>
+<ng-template #itemTemplate let-label="label" let-i="i">
+    <li
+        data-tag="menu-item"
+        [attr.data-index]="i"
+        [attr.disabled]="disabled"
+        [ngClass]="['normal-item-state', disabled ? 'is-disabled' : '']"
+    >
+        <div class="fd-menu__addon-before" *ngIf="icon; else noIcon">
+            <span [ngClass]="[icon]" data-tag="menu-item__icon-before" [attr.title]="icon"> </span>
+        </div>
+        <div class="fd-menu__label-truncation">
+            <span
+                class="fd-menu__item"
+                [ngStyle]="{ width: getItemWidth() }"
```

Comment:
Calling `getItemWidth()` within the property binding may cause performance issues. It is better to assign a variable here e.g. `[ngStyle]="{ width: finalItemWidth }"`.

And then in your component code, update the variable in the `OnChanges` cycle:

```ts
public finalItemWidth = '';

ngOnChanges(changes: SimpleChanges) {
  if (changes.itemWidth || changes.item) {
    this.finalItemWidth = this.getItemWidth();
  }
}

Reply:
done